apollo-federation/src/supergraph/mod.rs

```diff
@@ -1696,6 +1698,140 @@ lazy_static! {
     };
 }
 
+trait CompositeType {
```

Comment:
I like this, and I think we (as in the whole team) should look into applying this pattern across the crate. Just a note for the future.